### PR TITLE
Warn on identical autoload keys from two different libraries

### LINKED ISSUE #13431: Add more information to TInterpreter::ReadRootmapFile warnings
### Feature description

In my current setup I get warnings of the kind
  `Warning in <TInterpreter::ReadRootmapFile>: class  podio::CollectionBase found in libpodioDict.so  is already in libpodio.so`

For easier debugging it would be nice if the warning would contain the name of the rootmap file causing the issue.

### Alternatives considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dpiparo**
# This Pull request:

## Changes or fixes:
It improves the warnings of ROOT when two identical autoload keys are found for two different libraries. 

## Checklist:

- [v] tested changes locally
- [ ] updated the docs (if necessary)

This PR fixes #13431

### CODE REVIEW

**Comment by dpiparo** [comment]
See roottest PR https://github.com/root-project/roottest/pull/1001 for the test

File: core/metacling/src/TCling.cxx [hunk 1]
```diff
@@ -5549,25 +5547,31 @@ int TCling::ReadRootmapFile(const char *rootmapfile, TUniqueString *uniqueString
...
          TEnvRec *isThere = fMapfile->Lookup(keyname);
          if (isThere) {
             if (lib_name != isThere->GetValue()) { // the same key for two different libs
                if (firstChar == 'n') {
                   if (gDebug > 3)
-                     Info("ReadRootmapFile", "namespace %s found in %s is already in %s", keyname, lib_name.c_str(),
-                          isThere->GetValue());
+                     Info("ReadRootmapFile",
+                          "While processing %s, namespace %s was found to be associated to %s although it is already "
+                          "available for %s",
```
Comment (pcanal) [inline core/metacling/src/TCling.cxx:5558]:
```suggestion
                          "While processing %s, namespace %s was found to be associated to %s although it is already "
                          "associated to %s",
```

File: core/metacling/src/TCling.cxx [hunk 2]
```diff
@@ -5549,25 +5547,31 @@ int TCling::ReadRootmapFile(const char *rootmapfile, TUniqueString *uniqueString
...
                } else if (firstChar == 'h') { // it is a header: add the libname to the list of libs to be loaded.
                   lib_name += " ";
                   lib_name += isThere->GetValue();
                   fMapfile->SetValue(keyname, lib_name.c_str());
                } else if (!TClassEdit::IsSTLCont(keyname)) {
-                  Warning("ReadRootmapFile", "%s %s found in %s is already in %s", line.substr(0, keyLen).c_str(),
-                          keyname, lib_name.c_str(), isThere->GetValue());
+                  Warning("ReadRootmapFile",
+                          "While processing %s, %s %s was found to be associated to %s although it is already "
+                          "availbale for %s",
```
Comment (pcanal) [inline core/metacling/src/TCling.cxx:5567]:
```suggestion
                          "associated to %s",
```

**Comment by pcanal** [comment]
This PR could also be used to update the reference file for [roottest_root_core_recursive_recursivewarning](https://lcgapp-services.cern.ch/root-jenkins/job/root-pullrequests-build/182593/testReport/projectroot.roottest.root.core/recursive/roottest_root_core_recursive_recursivewarning/)

**Comment by dpiparo** [comment]
@phsft-bot build

**Comment by dpiparo** [comment]
@phsft-bot build

**Pull request merged**